src/common/OneDSLoggerTelemetry/oneDSLogger.ts

```diff
@@ -324,22 +325,24 @@ export class OneDSLogger implements ITelemetryLogger{
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     private populateVscodeWebAttributes(envelope: any) {
-        if (envelope.data.eventName ==  telemetryEventNames.WEB_EXTENSION_INIT_QUERY_PARAMETERS){
-            OneDSLogger.userInfo.tid= JSON.parse(envelope.data.eventInfo).tenantId;
-            OneDSLogger.userRegion = JSON.parse(envelope.data.eventInfo).geo;
-            OneDSLogger.contextInfo.orgId = JSON.parse(envelope.data.eventInfo).orgId;
-            OneDSLogger.contextInfo.portalId = JSON.parse(envelope.data.eventInfo).portalId;
-            OneDSLogger.contextInfo.websiteId = JSON.parse(envelope.data.eventInfo).websiteId;
-            OneDSLogger.contextInfo.dataSource = JSON.parse(envelope.data.eventInfo).dataSource;
-            OneDSLogger.contextInfo.schema = JSON.parse(envelope.data.eventInfo).schema;
-            OneDSLogger.contextInfo.correlationId = JSON.parse(envelope.data.eventInfo).referrerSessionId;
-            OneDSLogger.contextInfo.referrer = JSON.parse(envelope.data.eventInfo).referrer;
-            OneDSLogger.contextInfo.envId = JSON.parse(envelope.data.eventInfo).envId;
-            OneDSLogger.contextInfo.referrerSource = JSON.parse(envelope.data.eventInfo).referrerSource;
-            OneDSLogger.contextInfo.orgGeo = JSON.parse(envelope.data.eventInfo).orgGeo;
+        if (envelope.data.eventName == telemetryEventNames.WEB_EXTENSION_INIT_QUERY_PARAMETERS) {
+            const eventInfo = JSON.parse(envelope.data.eventInfo);
+
+            OneDSLogger.userInfo.tid = eventInfo.tenantId;
+            OneDSLogger.userRegion = eventInfo.geo ? geoMappingsToAzureRegion[eventInfo.geo.toLowerCase()].geoName : eventInfo.geo;
```

Comment:
What are we doing here?

If eventInfo.geo doesn't exist, we still end up logging that? In such a scenario what would be logged here?

Reply:
Quite unlikely eventInfo.geo will have undefined values because in Studio it is always defined. In any scenario it is undefined we should log empty

<img width="361" alt="image" src="https://github.com/microsoft/powerplatform-vscode/assets/97606074/7f64b10f-0d39-4895-978e-c83ddb273280">